contentpacks/utils.py

```diff
@@ -707,12 +707,20 @@ def remove_nonexistent_assessment_items_from_exercises(node_data: list, assessme
 
 def remove_exercises_without_assessment_items(node_data):
     # TODO: This function must be refactor to use the yield generators.
+    # TODO: Create another function to remove duplicate youtubeIds in node data.
+    yotube_ids = []
     new_node_data = []
     for node in node_data:
-        if node["kind"] != NodeType.exercise:
-            new_node_data.append(node)
-        else:
+        if node["kind"] == NodeType.exercise:
             assessment_items = node["all_assessment_items"]
             if assessment_items:
```

Comment:
Is the `assessment_items` variable necessary here?  Maybe you can just use the `if node.get("all_assessment_items")` statement instead?